core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -1358,17 +1367,241 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
+    assertEquals(offset, offsetFetchResponse.responseData(group).get(tp).offset)
+  }
+
+  @Test
+  def testOffsetFetchMultipleGroupsAuthorization(): Unit = {
+    val groupOne = "group1"
+    val groupOneResource = new ResourcePattern(GROUP, groupOne, LITERAL)
+    val groupTwo = "group2"
+    val groupTwoResource = new ResourcePattern(GROUP, groupTwo, LITERAL)
+    val groupThree = "group3"
+    val groupThreeResource = new ResourcePattern(GROUP, groupThree, LITERAL)
+    val groupFour = "group4"
+    val groupFourResource = new ResourcePattern(GROUP, groupFour, LITERAL)
+    val groupFive = "group5"
+    val groupFiveResource = new ResourcePattern(GROUP, groupFive, LITERAL)
+
+    val topic1 = "topic1"
+    val topic1List = singletonList(new TopicPartition(topic1, 0))
+    val topicOneResource = new ResourcePattern(TOPIC, topic1, LITERAL)
+    val topic2 = "topic2"
+    val topic1And2List = util.Arrays.asList(
+      new TopicPartition(topic1, 0),
+      new TopicPartition(topic2, 0),
+      new TopicPartition(topic2, 1))
+    val topicTwoResource = new ResourcePattern(TOPIC, topic2, LITERAL)
+    val topic3 = "topic3"
+    val allTopicsList = util.Arrays.asList(
+      new TopicPartition(topic1, 0),
+      new TopicPartition(topic2, 0),
+      new TopicPartition(topic2, 1),
+      new TopicPartition(topic3, 0),
+      new TopicPartition(topic3, 1),
+      new TopicPartition(topic3, 2))
+    val topicThreeResource = new ResourcePattern(TOPIC, topic3, LITERAL)
+
+    // create group to partition map to build batched offsetFetch request
+    val groupToPartitionMap = new util.HashMap[String, util.List[TopicPartition]]()
+    groupToPartitionMap.put(groupOne, topic1List)
+    groupToPartitionMap.put(groupTwo, topic1And2List)
+    groupToPartitionMap.put(groupThree, allTopicsList)
+    groupToPartitionMap.put(groupFour, null)
+    groupToPartitionMap.put(groupFive, null)
+
+    createTopic(topic1)
+    createTopic(topic2, numPartitions = 2)
+    createTopic(topic3, numPartitions = 3)
+
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), groupOneResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), groupTwoResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), groupThreeResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), groupFourResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), groupFiveResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), topicOneResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), topicTwoResource)
+    addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WildcardHost, READ, ALLOW)), topicThreeResource)
```

Comment:
Looks like they are are all the same with different resources, better to put them in a collection and call `addAndVerifyAcls` over the collection